provider/secrets_manager_provider.go

```diff
@@ -261,15 +262,15 @@ func NewSecretsManagerProviderWithClients(clients ...SecretsManagerClient) *Secr
 	}
 }
 
-func NewSecretsManagerProvider(awsSessions []*session.Session, regions []string) *SecretsManagerProvider {
-	var clients []SecretsManagerClient
-	for i, awsSession := range awsSessions {
+func NewSecretsManagerProvider(configs []aws.Config, regions []string) *SecretsManagerProvider {
+	var secretsManagerClients []SecretsManagerClient
+	for i, cfg := range configs {
 		client := SecretsManagerClient{
-			Region:     *awsSession.Config.Region,
-			Client:     secretsmanager.New(awsSession, aws.NewConfig().WithRegion(regions[i])),
+			Region:     regions[i],
+			Client:     secretsmanager.NewFromConfig(cfg),
```

Comment:
I'll have to double check, but I may need to override the `cfg.Region = regions[i]`.